### PR TITLE
arithmetic operations on curves

### PR DESCRIPTION
**Pull request by stonneau**
We can now add, subtract and multiply / divide by a scalar polynomial and Bezier curves.
These operations require same degree curves for Bezier curves, thus an elevate method has been added,
which allows to transform a Bezier curve into an equivalent curve, one degree higher.

Additionally, the cross product method has been added for both classes.
A todo is that while the simplification is done for polynomials (when the higher degrees are worth 0), it is not trivial to achieve for Bezier curves. I still don't know whether it would be best to convert back to the monomial basis, compute the cross product and then convert back to the Bernstein basis. Of course the conversion itself is slower, but in the long run it makes a difference in an optimisation problem for instance.

Python bindings added as well.

### CODE REVIEW

File: include/curves/cross_implementation.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/**
+ * \file cubic_hermite_spline.h
+ * \brief class allowing to create a cubic hermite spline of any dimension.
+ * \author Justin Carpentier <jcarpent@laas.fr> modified by Jason Chemin <jchemin@laas.fr>
+ * \date 05/2019
```
Comment (pFernbach) [inline include/curves/cross_implementation.h]: Header/license is not updated

File: include/curves/linear_variable.h [hunk 2]
```diff
@@ -142,6 +167,7 @@ struct linear_variable : public serialization::Serializable {
   matrix_x_t B_;
   vector_x_t c_;
   bool zero;
+  static const double MARGIN;
```
Comment (pFernbach) [inline include/curves/linear_variable.h]: There was already this const definition in Bezier, and now it's required here and in polynomial. Shouldn't we move this in an higher level file to have only one definition for the whole package ?

File: include/curves/polynomial.h [hunk 3]
```diff
@@ -42,6 +42,7 @@ struct polynomial : public curve_abc<Time, Numeric, Safe, Point> {
   typedef Eigen::Ref<coeff_t> coeff_t_ref;
   typedef polynomial<Time, Numeric, Safe, Point, T_Point> polynomial_t;
   typedef typename curve_abc_t::curve_ptr_t curve_ptr_t;
+  static const double MARGIN;
```
Comment (pFernbach) [inline include/curves/polynomial.h]: See comment on linear_variable.

**Review by pFernbach — APPROVED** [review]
Thanks for this contribution. I have minor remarks, otherwise if the CI pass I will merge.

https://gitlab.laas.fr/pfernbac/curves/-/pipelines/11475

**Comment by pFernbach** [comment]
There is some errors in the CI: https://gitlab.laas.fr/pfernbac/curves/-/jobs/93744 
I am not sure if it's related to this PR. I will try to check the previous version.

**Comment by stonneau** [comment]
Hi there I realised this was never merged.
Is the CI still not passing ? Did you have a chance to investigate @pFernbach 
thanks a lot
Steve

**Comment by pFernbach** [comment]
I restarted the CI and it's not passing on ubuntu 16.04 because of a the new unit tests you added : https://gitlab.laas.fr/loco-3d/curves/-/jobs/93880

Looks like the macro you are using did not exist in 16.04, could you check if you can achieve the same with something backward compatible ? Otherwise check with @nim65s

**Comment by stonneau** [comment]
@pFernbach  Should be good now.

**Pull request merged**